modules/data.atmosphere/R/temporal.downscaling.R

```diff
@@ -100,85 +104,116 @@ cfmet.downscale.subdaily <- function(subdailymet, output.dt = 1) {
     }
   }
 
-  downscaled.result <- cbind(new.date, data.table::as.data.table(downscaled.result))
+  downscaled.result <- cbind(new.date, as.data.frame(downscaled.result))
 } # cfmet.downscale.subdaily
 
 
+#' Internal helper to downscale a single row from a daily file
+#'
+#' @param df one row from dailymet
+#' @param tseq vector of hours at which to estimate
+#' @param lat latitude
+#'
+#' @return df with one row for each hour in `tseq`
+#'
+downscale_one_cfmet_day <- function(df, tseq, lat) {
+  if (nrow(df) != 1) {
+    PEcAn.logger::logger.error("df must be a one-row dataframe")
+  }
+  if (length(unique(diff(tseq))) != 1) {
+    PEcAn.logger::logger.error("tseq has uneven intervals")
+  }
+
+  n <- length(tseq)
+
+  light <- lightME(DOY = df$doy, t.d = tseq, lat = lat) %>%
+    as.data.frame() %>%
+    dplyr::mutate(
+      Itot = .data$I.dir + .data$I.diff,
+      resC2 = (.data$Itot - min(.data$Itot)) / max(.data$Itot))
+
+  rhscale <- (cos(2 * pi * (tseq - 10) / n) + 1) / 2
+
+  data.frame(
+    hour = tseq,
+    solarR = rep(
+        df$surface_downwelling_shortwave_flux_in_air * 2.07 * 10^5 / 36000,
+        each = n)
+      * light$resC2,
+    # Note: When dt >= 6, all times get the same T prediction
+    Temp = df$tmin +
+      (sin(2 * pi * (tseq - 10) / n) + 1) /
+      2 * (df$tmax - df$tmin),
+    relative_humidity = df$rhmin + rhscale * (df$rhmax - df$rhmin),
+    # TODO: Why do we divide by n?
+    #   isn't precipitation_flux already an intensity?
+    precipitation_flux = rep(df$precipitation_flux / n, each = n),
+    wind = rep(df$wind_speed, each = n)) %>%
+  dplyr::mutate(
+    # TODO: Computation of solarR above already multiplies by resC2.
+    #   Is multiplying it again here really correct?
+    #   That's how the old data.table version did it
+    #   (once when computing `solarR` and again when computing `SolarR`),
+    #   so keeping it until proven wrong.
+    downwelling_photosynthetic_photon_flux = .data$solarR * light$resC2)
```

Comment:
I don't know radiation calculations well enough to know if this is correct or not, just seemed odd to me especially since they're done in two separate steps. Can a met expert weigh in here, please?

Reply:
(see line 133 for the first multiplication by `resC2`)

Reply:
(And see deleted lines 136 and 138 of the for the same calculations as implemented before this PR)